extras/controllerClient/x86/example_c_sharp.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace NVAccess
+{
+    /// <summary>
+    /// C# wrapper class for the NVDA controller client DLL
+    /// </summary>
+    public static class NVDA
+    {
+        [DllImport("nvdaControllerClient32.dll")]
+        private static extern int nvdaController_testIfRunning();
+
+        [DllImport("nvdaControllerClient32.dll", CharSet = CharSet.Unicode)]
+        private static extern int nvdaController_speakText(StringBuilder text);
+
+        [DllImport("nvdaControllerClient32.dll")]
+        private static extern int nvdaController_cancelSpeech();
+
+        [DllImport("nvdaControllerClient32.dll")]
+        private static extern int nvdaController_brailleMessage(StringBuilder text);
+        
+        /// <summary>
+        /// Checks to see if NVDA is running
+        /// </summary>
+        public static bool IsRunning
+        {
+            get
+            {
+                return nvdaController_testIfRunning() == 0;
+            }
+        }
+
+        /// <summary>
+        /// Tells NVDA to speak a string of text
+        /// </summary>
+        /// <param name="text">The text to speak</param>
+        /// <param name="interrupt">If true, NVDA will stop whatever it's saying at th emoment and immediately speak the string</param>
+        public static void Say(string text, bool interrupt = false)
+        {
+            if (interrupt)
+                Stop();
+
+            StringBuilder sb = new StringBuilder(text.Length);
+            sb.Append(text);
+            nvdaController_speakText(sb);
+        }
+
+        /// <summary>
+        /// Tells NVDA to braille a string of text (UNTESTED)
```

Comment:
I assume this is labelled as untested because you do not have a braille display? To test braille output consider: https://github.com/nvaccess/nvda/wiki/Testing-braille-output-without-a-braille-display

Reply:
@DojoMike did you test this? If you are happy with it, please remove the (UNTESTED) comment and we can progress this issue. Thanks!